### PR TITLE
Add get refinement functions 

### PR DESCRIPTION
**Pull request by wwkong**
Adds the functions `getAnisotropicRefinement()` and `getSurplusRefinement()` to the Python interface.

### CODE REVIEW

File: InterfacePython/TasmanianSG.py [hunk 1]
```diff
@@ -1525,6 +1525,13 @@ def setAnisotropicRefinement(self, sType, iMinGrowth, iOutput, liLevelLimits = [
             sType = bytes(sType, encoding='utf8')
         pLibTSG.tsgSetAnisotropicRefinement(self.pGrid, c_char_p(sType), iMinGrowth, iOutput, pLevelLimits)
 
+    def getAnisotropicRefinement(self, sType, iMinGrowth, iOutput, liLevelLimits = []):
+        '''
+        Calls setAnistropicRefinement() on the inputs and then getNeededPoints().
+        '''
+        self.setAnisotropicRefinement(sType, iMinGrowth, iOutput, liLevelLimits = [])
```
Comment (mkstoyanov) [inline InterfacePython/TasmanianSG.py:1532]: What?! That's now how you pass in parameters in Python.

Also, have them implemented in also in C++.

File: InterfacePython/TasmanianSG.py [hunk 2]
```diff
@@ -1654,6 +1661,13 @@ def setSurplusRefinement(self, fTolerance, iOutput, sCriteria = "", liLevelLimit
                 sCriteria = bytes(sCriteria, encoding='utf8')
             pLibTSG.tsgSetLocalSurplusRefinement(self.pGrid, c_double(fTolerance), c_char_p(sCriteria), iOutput, pLevelLimits, pScaleCorrection)
 
+    def getSurplusRefinement(self, fTolerance, iOutput, sCriteria = "", liLevelLimits = [], llfScaleCorrection = []):
+        '''
+        Calls setSurplusRefinement() on the inputs and then getNeededPoints().
+        '''
+        self.setSurplusRefinement(fTolerance, iOutput, sCriteria = "", liLevelLimits = [], llfScaleCorrection = [])
```
Comment (mkstoyanov) [inline InterfacePython/TasmanianSG.py:1668]: Lazy copy-paste, don'w do that!

**Review by mkstoyanov — CHANGES_REQUESTED** [review]
Focus!

File: InterfacePython/TasmanianSG.py (hunk 1 above)
Comment (wwkong) [reply InterfacePython/TasmanianSG.py]: Fixed. Will do the C++ functions now.

File: SparseGrids/TasmanianSparseGrid.hpp [hunk 3]
```diff
@@ -1200,6 +1207,13 @@ class TasmanianSparseGrid{
     void setSurplusRefinement(double tolerance, int output, std::vector<int> const &level_limits);
     //! \brief Overload that uses array for the level limits.
     void setSurplusRefinement(double tolerance, int output, const int *level_limits = nullptr);
+    /*!
+     * \brief Call setSurplusRefinement() for Sequence and Global grids with a sequence rule and then getNeededPoints().
+     */
+    std::vector<double> getSurplusRefinement(double tolerance, int output, std::vector<int> const &level_limits) {
```
Comment (mkstoyanov) [inline SparseGrids/TasmanianSparseGrid.hpp:1213]: This is OK. Just add it to the Doxygen at the top of the file where different methods are split into categories.

Comment (wwkong) [reply SparseGrids/TasmanianSparseGrid.hpp:1214]: Done.

**Review by mkstoyanov — APPROVED** [review]

**Pull request merged**